lib/ts/recipe/session/api/implementation.ts

```diff
@@ -82,3 +98,83 @@ export default function getAPIInterface(): APIInterface {
         },
     };
 }
+
+/**
+ *  This function addresses an edge case where changing the cookieDomain config on the server can
+ *  lead to session integrity issues. For instance, if the API server URL is 'api.example.com'
+ *  with a cookie domain of '.example.com', and the server updates the cookie domain to 'api.example.com',
+ *  the client may retain cookies with both '.example.com' and 'api.example.com' domains.
+ *
+ *  Consequently, if the server chooses the older cookie, session invalidation occurs, potentially
+ *  resulting in an infinite refresh loop. To fix this, users are asked to specify "olderCookieDomain" in
+ *  the config.
+ *
+ * This function checks for multiple cookies with the same name and clears the cookies for the older domain
+ */
+function clearSessionCookiesFromOlderCookieDomain({
+    req,
+    res,
+    config,
+    userContext,
+}: {
+    req: BaseRequest;
+    res: BaseResponse;
+    config: TypeNormalisedInput;
+    userContext: UserContext;
+}): boolean {
+    let didClearCookies = false;
+    const cookieString = req.getHeaderValue("cookie");
+
+    if (cookieString === undefined || config.olderCookieDomain === undefined) {
+        return didClearCookies;
+    }
+
+    const cookies = parseCookieStringFromRequestHeader(cookieString);
+
+    const tokenTypes: TokenType[] = ["access", "refresh"];
+    for (const token of tokenTypes) {
+        const cookieName = getCookieNameFromTokenType(token);
+
+        if (cookies[cookieName] && cookies[cookieName].length > 1) {
+            logDebugMessage(
+                `clearDuplicateSessionCookies: Clearing duplicate ${cookieName} cookie with domain ${config.olderCookieDomain}`
+            );
+            setToken(
+                { ...config, cookieDomain: config.olderCookieDomain },
+                res,
+                token,
+                "",
+                0,
+                "cookie",
+                req,
+                userContext
+            );
+            didClearCookies = true;
+        }
+    }
+
+    return didClearCookies;
+}
+
+// This function is required because cookies library (and most of the popular libraries in npm)
+// does not support parsing multiple cookies with the same name.
+function parseCookieStringFromRequestHeader(cookieString: string): Record<string, string[]> {
```

Comment:
not necessary to change, but since we will not be using the value we do not need to parse the header, it could be something like a regexp check

Reply:
This could prove useful in the future, and since I've already tested this function, I'm choosing to keep it as is.